lib/console_web/controllers/gateway_controller.ex

```diff
@@ -61,6 +61,34 @@ defmodule ConsoleWeb.GatewayController do
     end
   end
 
+  def register(conn, %{"OUI" => oui, "nonce" => nonce, "gateway" => gateway}) do
+    # receive the following from router:
+    #   nonce and OUI self explanatory
+    #   gateway["id"] is 32 bit gateway ID
```

Comment:
Should we randomly generate a 32 bit random gateway id? 

Do we convert the 32 bit binary into a list and send it over http? So what router would send back is a list that we can use list_to_binary on

Reply:
i'd imagine we'd pass it as a string since that's a little more compatible with json apis

Reply:
and `gateway["id"]` is the ID in OUI/ID right? not the gateway row's primary key id?

Reply:
yup, have yet to make that table, was figuring out how to generate the binary first. 

Will probably use :crypto.strong_rand_bytes(4)